R/DiseasyActivity.R

```diff
@@ -549,17 +567,38 @@ DiseasyActivity <- R6::R6Class( # nolint: object_name_linter
 
     activity_types = c("home", "work", "school", "other"),
 
-    add_activities = function(x) {
-      tmp_units <- private$activity_units[x] # List of units to add
-      obj <- list()
-      for (type in private$activity_types) {
-        obj[[type]] <- rep(0, private$n_age_groups)
-        for (i in seq_along(tmp_units)) {
-          obj[[type]] <- obj[[type]] + tmp_units[[i]][[type]] * tmp_units[[i]][["risk"]]
+    # Risk-weighted activity
+    # @description
+    #   This function computes the risk-weighted activity of the given activities across
+    #   the `activity_types`
+    # @param activities (`character`)\cr
+    #   A vector of activities to add together
+    # @return
+    #   A list of depth two: value[[type]][[age_group_activity]]
+    add_activities = function(activities) {
+
+      # Extract the relevant activity units
+      activity_unit_subset <- private$activity_units[activities]
+
+      # For each activity type, we
+      # 1) multiply by activity by the associated risk
+      # 2) sum the risk-weighted activities stratified by age-group
+      # 3) set human readable names for the age_groups
+      # (Activity is expressed in 5-year age groups, we name our activity vector accordingly)
+      risk_weighted_activity <- purrr::map(
+        private$activity_types,
+        \(type) {
+          activity_unit_subset |>
+            purrr::map(~ purrr::pluck(., type) * purrr::pluck(., "risk")) |>
```

Comment:
I might misinterpret, but is there an inconsistency when not visually binding `purrr::pluck(., type)`, but then visually binding `purrr::pluck(., "risk")` ?

Reply:
Notice that this is an anonymous function ( it starts with `\(type) {` ) so within this scope, `type` is bound.

That is, `type` does not match a column named "type" in `.` but rather the value stored in `type` matches columns of `.`

Reply:
Roger that. I dont have that much experience with anonymous functions!